### PR TITLE
#40 readlink and greadlink switch on OS basis

### PR DESCRIPTION
**Pull request by MuschPusch**

### CODE REVIEW

File: template/configure.sh [hunk 1]
```diff
@@ -1,8 +1,15 @@
 #!/bin/bash
 . .git/hooks/hook_switcher.sh
 
+if [[ "$OSTYPE" == *darwin* ]]
+then
+  READLINK_CMD='greadlink'
+else
+  READLINK_CMD='readlink'
+fi
```
Comment (greg0ire) [inline template/configure.sh]: Please use tabs for indention.

File: template/configure.sh [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+ 
 readonly PROGNAME=$(basename "$0")
-readonly PROGDIR=$(readlink -m "$(dirname "$0")")
+readonly PROGDIR=$($READLINK_CMD -m "$(dirname "$0")")
```
Comment (greg0ire) [inline template/configure.sh]: Please quote the new variable you introduced.

**Comment by MuschPusch** [comment]
Problem is that readlink exists on os X but it expects different parameters. greadlink is part of core utils which needs to get installed.

**Comment by greg0ire** [comment]
Looks like `greadlink` is [the way to go](http://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac). Can you add a § in the documentation, under [installation](https://github.com/greg0ire/git_template/blob/master/docs/index.md#installation) about that ?

**Comment by greg0ire** [comment]
Also, your PR has two commits, please use `git rebase --interactive master` to fix this.

**Comment by greg0ire** [comment]
@MuschPusch Did I frighten you with my comments?

**Comment by MuschPusch** [comment]
No all fine but it's busy day today at work. i will try work on this asap :)

**Comment by greg0ire** [comment]
Ok great!

**Comment by greg0ire** [comment]
ping @MuschPusch :P

File: template/configure.sh (hunk 2 above)
Comment (METAJIJI) [inline template/configure.sh]: May be most compatible variant:
```
readonly PROGDIR="$(cd "$(dirname "$0")"; pwd)"
```

Comment (greg0ire) [inline template/configure.sh]: Maybe, but then it should be followed with `cd -`, right?

Comment (METAJIJI) [inline template/configure.sh]: `cd -` is a shell feature, and it is unlikely but can not work.
It will be safer if you use the option with `dirname`

Comment (METAJIJI) [inline template/configure.sh]: Also `cd -` return `OLDPWD` It's not what we need :)))

Comment (greg0ire) [inline template/configure.sh]: Would you like to carry this PR?

**Comment by greg0ire** [comment]
Closing in favor of #49

**Pull request closed without merging**